yesod-core/src/Yesod/Core/HandlerContents.hs

```diff
@@ -0,0 +1,37 @@
+module Yesod.Core.HandlerContents
```

Comment:
I broke this type definition out into a separate file to avoid a circular dependency. Maybe someone else can see a better way to solve this?

Reply:
I think this is fine - just make sure `HandlerContents` is being properly re-exported from the original module in which it is defined.

You'll need to modify the `yesod-core.cabal` file and add this as either an `exposed-modules` (which makes it public so you'll want to add a CHANGELOG entry + some haddocks here) or `other-modules` (in which case it is private and you don't need to do quite so much ((though having an `@since` tag in some moduel docs is nice for the developer trawlin code)))